OpenRA.Mods.Common/Activities/Enter.cs

```diff
@@ -36,13 +37,22 @@ enum EnterState { ApproachingOrEntering, Inside, Exiting, Done }
 		Activity inner;
 		bool firstApproach = true;
 
-		protected Enter(Actor self, Actor target, EnterBehaviour enterBehaviour, int maxTries = 1, bool repathWhileMoving = true)
+		/// <summary>
+		/// Make an actor enter the target actor.
+		/// </summary>
+		/// <param name="enterRange">
+		///		When the distance between the target actor and self is closer than or equal to this range,
+		///		self will enter the target. Non-zero range allows self to enter a moving target.
+		/// </param>
+		protected Enter(Actor self, Actor target, EnterBehaviour enterBehaviour, WDist enterRange,
```

Comment:
I think it would be a more elegant solution if you made it `enterRange = WDist.Zero`, then you could revert the ctor changes to the other traits/activities (except EnterTransport).

Reply:
I tried that at first, but it refused to compile because WDist is not nullable.

Reply:
You can get the equivalent behaviour by defining an alternative constructor that doesn't accept the argument.